src/postgres/src/backend/utils/misc/pg_yb_utils.c

```diff
@@ -1474,3 +1476,99 @@ yb_hash_code(PG_FUNCTION_ARGS)
 	uint16_t hashed_val = YBCCompoundHash(arg_buf, total_bytes);
 	PG_RETURN_UINT16(hashed_val);
 }
+
+/*---------------------------------------------------------------------------*/
+/* Deterministic DETAIL order                                                */
+/*---------------------------------------------------------------------------*/
+
+static int yb_detail_sort_comparator(const void *a, const void *b)
+{
+	return strcmp(*(const char **) a, *(const char **) b);
+}
+
+typedef struct {
+	char **lines;
+	int length;
+} DetailSorter;
+
+void detailSorterFromList(DetailSorter *v, List *litems, int capacity)
+{
+	v->lines = (char **)palloc(sizeof(char *) * capacity);
+	v->length = 0;
+	ListCell *lc;
+	foreach (lc, litems)
+	{
+		if (v->length < capacity)
+		{
+			v->lines[v->length++] = (char *)lfirst(lc);
+		}
+	}
+}
+
+char **detailSorterLinesSorted(DetailSorter *v)
+{
+	qsort(v->lines, v->length,
+		sizeof (const char *), yb_detail_sort_comparator);
+	return v->lines;
+}
+
+void detailSorterFree(DetailSorter *v)
+{
+	pfree(v->lines);
+}
+
+char *yb_detail_sorted(char *input)
+{
+	if (input == NULL)
+		return input;
+
+	// this delimiter is hard coded in backend/catalog/pg_shdepend.c,
+	// inside of the storeObjectDescription function:
+	char delimiter[2] = "\n";
+
+	// init stringinfo used for concatenation of the output:
+	StringInfoData s;
+	initStringInfo(&s);
+
+	// this list stores the non-empty tokens, extra counter to know how many:
+	List *line_store = NIL;
+	int line_count = 0;
+
+	char *token;
+	token = strtok(input, delimiter);
+	while (token != NULL)
+	{
+		if (strcmp(token, "") != 0)
+		{
+			line_store = lappend(line_store, token);
+			line_count++;
+		}
+		token = strtok(NULL, delimiter);
+	}
+
+	DetailSorter sorter;
+	detailSorterFromList(&sorter, line_store, line_count);
+
+	if (line_count == 0)
+	{
+		// put the original input in:
+		appendStringInfoString(&s, input);
+	}
+	else
+	{
+		char ** sortedLines = detailSorterLinesSorted(&sorter);
```

Comment:
nit: no space after `**` and underscore case for `sorted_lines`

Reply:
https://github.com/yugabyte/yugabyte-db/pull/9650/commits/96eeb6b8a7253ae54952896fb54cb41ea21db32d